### PR TITLE
Refactor AddPerson and Edit Commands

### LINKED ISSUE #13: As a property agent, I want to be able to edit my database of buyers on the go, without a need for an internet connection.

### PR DESCRIPTION
**Pull request by ChryslineLim**
Fixes #13, #16, #76, #78, #86

### CODE REVIEW

File: src/main/java/seedu/address/model/property/Properties.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package seedu.address.model.role;
+package seedu.address.model.property;
```
Comment (chngchngchng) [inline src/main/java/seedu/address/model/property/Properties.java:1]: We may not need this class since it was originally used to represent a list of `Properties` that a seller is in charge of - but we no longer make use of a `Seller` class anymore.

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 2]
```diff
@@ -103,8 +116,29 @@ public Person toModelType() throws IllegalValueException {
         }
         final Address modelAddress = new Address(address);
 
+        if (priceRange == null) {
```
Comment (chngchngchng) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:119]: This code seems redundant, since the above conditions to get priceRange will set it to `""` (the empty string) if the `Optional` contains `null`.

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 3]
```diff
@@ -103,8 +116,29 @@ public Person toModelType() throws IllegalValueException {
...
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    PriceRange.class.getSimpleName()));
+        }
+        if (!priceRange.isEmpty() && !PriceRange.isValidPriceRange(priceRange)) {
+            throw new IllegalValueException(PriceRange.MESSAGE_CONSTRAINTS);
+        }
+        final PriceRange modelPriceRange = priceRange.isEmpty() ? null : new PriceRange(priceRange);
+
+        if (desiredCharacteristics == null) {
```
Comment (chngchngchng) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:128]: Same for this null check as above

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 4]
```diff
@@ -69,20 +63,11 @@ public PersonCard(Person person, int displayedIndex) {
...
-
-        if (person.isSeller()) {
-            sellerRole.setText(person.isSeller() ? "Seller" : "");
-            person.getSellerProperties()
-                    .getPropertyList()
-                    .forEach(pr -> sellerProperties.getChildren()
-                            .add(new Label(pr.toString())));
-        }
+        priceRange.setText("Budget: " + person.getPriceRange()
+                .map(PriceRange::toString).orElse("Budgt: Not Specified"));
```
Comment (chngchngchng) [inline src/main/java/seedu/address/ui/PersonCard.java:67]:
```suggestion
                .map(PriceRange::toString).orElse("Budget: Not Specified"));
```

File: src/main/resources/view/PersonListCard.fxml [hunk 5]
```diff
@@ -31,11 +31,8 @@
       <Label fx:id="phone" styleClass="cell_small_label" text="\$phone" />
       <Label fx:id="address" styleClass="cell_small_label" text="\$address" />
       <Label fx:id="email" styleClass="cell_small_label" text="\$email" />
-      <Label fx:id="buyerRole" styleClass="cell_small_label" text="\$buyerRole" />
-      <Label fx:id="buyerRange" styleClass="cell_small_label" text="\$buyerRange" />
-      <Label fx:id="buyerCharacteristics" styleClass="cell_small_label" text="\$buyerCharacteristics" />
-      <Label fx:id="sellerRole" styleClass="cell_small_label" text="\$sellerRole" />
-      <FlowPane fx:id="sellerProperties"/>
+      <Label fx:id="priceRange" styleClass="cell_small_label" text="\$priceRange" />
+      <Label fx:id="desiredCharacteristics" styleClass="cell_small_label" text="\$desiredCharacteristics" />
```
Comment (chngchngchng) [inline src/main/resources/view/PersonListCard.fxml:35]: If the desired characteristics are extremely long, how will the Label look like?

**Review by chngchngchng — COMMENTED** [review]
Overall LGTM! Just some minor changes and edge cases to handle here and there.

File: src/main/java/seedu/address/model/pricerange/PriceRange.java [hunk 6]
```diff
@@ -45,6 +42,11 @@ public static boolean isValidPriceRange(String test) {
 
         boolean isValid = test.matches(VALIDATION_REGEX);
 
+        // to prevent out of bounds error below
+        if (!isValid) {
+            return false;
+        }
+
         String[] rangeValues = test.split("-");
         double leftValue = Float.parseFloat(rangeValues[0].trim());
```
[the remark this answers is not in the document]
Comment (ChryslineLim) [reply src/main/java/seedu/address/model/pricerange/PriceRange.java:51]: It doesn't cause it to crash or anything because I think parseFloat just parses it into an Infinity number if it's beyond the range, but I realised that it actually affects the accuracy a bit? Like when I tested 20000000000000 as the upper boundary, it displayed as 199999999981808 (something like that). So I changed it to parseDouble and it seems to work now!

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java (hunk 2 above)
Comment (ChryslineLim) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java:119]: I agree with you, I think it's a just-in-case check? Because for Name, Address, etc. they all checked for nullity too although those fields aren't even optional...

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 4 above)
Comment (ChryslineLim) [reply src/main/java/seedu/address/ui/PersonCard.java:67]: Thanks I've changed this!

**Review by jchilling — APPROVED** [review]
LGTM 👍
